pre_commit_hooks/loaderon_hooks/tests/general_hooks/check_branch_name_test.py

```diff
@@ -0,0 +1,37 @@
+import subprocess
+import sys
+import uuid
+
+import pytest
+
+from pre_commit_hooks.loaderon_hooks.general_hooks.check_branch_name import main
+
+
+@pytest.fixture(autouse=True)
+def clean_sys_argv():
+    sys.argv = []
+    yield
+
+
+def test_branch_name_ok():
+    new_branch_name = str(uuid.uuid4())
```

Comment:
why use a uuid?  I would just use a fixed value

oh, this is running on the repository itself -- yikes

ok I would make a fixture that creates a temporary git directory in a tempdir so you don't modify your work in place (potentially blowing it away!)

Reply:
there's a few tests in this repository that do something like that, check out the `check-added-large-files` tests